File: jvm/core/src/main/java/se/idkollen/client/models/VippsBackchannelAuthRequest.java
package se.idkollen.client.models;

import org.jspecify.annotations.Nullable;

/**
 * Request body for starting a Vipps MobilePay backchannel authentication session.
 */
public class VippsBackchannelAuthRequest {
    public String phone = "";
    public @Nullable Boolean requestSsn;
    public @Nullable Boolean requestEmail;
    public @Nullable Boolean requestAddress;
    public @Nullable String callbackUrl;
    public @Nullable String refId;

    public VippsBackchannelAuthRequest setPhone(String v) {
        phone = v;
        return this;
    }

    public VippsBackchannelAuthRequest setRequestSsn(@Nullable Boolean v) {
        requestSsn = v;
        return this;
    }

    public VippsBackchannelAuthRequest setRequestEmail(@Nullable Boolean v) {
        requestEmail = v;
        return this;
    }
    public VippsBackchannelAuthRequest setRequestAddress(@Nullable Boolean v) {
        requestAddress = v;
        return this;
    }

    public VippsBackchannelAuthRequest setCallbackUrl(@Nullable String v) {
        callbackUrl = v;
        return this;
    }

    public VippsBackchannelAuthRequest setRefId(@Nullable String v) {
        refId = v;
        return this;
    }
}

File: php/src/Models/Vipps/VippsBackchannelAuthRequest.php
<?php

declare(strict_types=1);

namespace Idkollen\Client\Models\Vipps;

readonly class VippsBackchannelAuthRequest
{
    public function __construct(
        public string $phone,
        public ?bool $requestSsn = null,
        public ?bool $requestEmail = null,
        public ?bool $requestAddress = null,
        public ?string $callbackUrl = null,
        public ?string $refId = null,
    ) {}
}
